src/codegen/properties/attribute.py

```diff
@@ -0,0 +1,281 @@
+        self._id  = id
+        self._is_enum = False
+        self._access = attribute_metadata["access"]
+        self._name = attribute_metadata["name"].lower()
+        self._resettable = attribute_metadata["resettable"]
+        self._type = attribute_metadata["type"]
+        self._ctypes_data_type = attribute_metadata["ctypes_data_type"]
+        self._python_data_type = attribute_metadata["python_data_type"]
+        self._description = attribute_metadata["description"]
+        self._has_explicit_read_buffer_size = attribute_metadata["has_explicit_read_buffer_size"]
+        self._bitfield_enum = attribute_metadata["bitfield_enum"]
+        self._is_list = attribute_metadata["is_list"]
+        self._calling_convention = attribute_metadata["calling_convention"]
+        self._c_function_name = attribute_metadata["c_function_name"]
+        if("is_object" in attribute_metadata): 
+            self._is_object = attribute_metadata["is_object"]
+        if("read_buffer_size" in attribute_metadata):
+            self._read_buffer_size = attribute_metadata["read_buffer_size"]
+        self._python_class_name = attribute_metadata["python_class_name"]
+        self._handle_parameters = []
+        self._object_constructor_params = []
+        if("handle_parameters" in attribute_metadata):
+            for name, parameter_data in attribute_metadata["handle_parameters"].items():
+                self._handle_parameters.append(Parameter(name, parameter_data))
+        if("object_has_factory" in attribute_metadata):
+            self._object_has_factory = attribute_metadata["object_has_factory"]
+        if("object_constructor_params" in attribute_metadata):
+            for name, parameter_data in attribute_metadata["object_constructor_params"].items():
+                self._object_constructor_params.append(Parameter(name, parameter_data))
+        if("has_explicit_write_buffer_size" in attribute_metadata):
+            self._has_explicit_write_buffer_size = attribute_metadata["has_explicit_write_buffer_size"]
+        if("enum" in attribute_metadata):
+            self._enum = attribute_metadata["enum"]
+            self._is_enum = True
+        if("object_type" in attribute_metadata):
+            self._object_type = attribute_metadata["object_type"]
+            
+    @property
+    def id(self):
+        """
+        Represents a unique integer value that represents an attribute.
+        This is the key for the attribute itself
+        """
+        return self._id
+
+
+    @property
+    def access(self):
+        """
+        Specifies if the attribute is read/write.
+        Supported Values
+        - `read`: The attribute is readable.(get)
+        - `write`: The attribute is writeable or can be set with new values.(set)
+        - `read-write`: The attribute can be both be read and written.(get and set)
+        This is used to decide the generation of getters and setters of a property representing the attribute.
```

Comment:
The rest of the code in this repo seems to use Google style docstrings, but these docstrings are not formatted that way.

See: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

> A docstring should be organized as a summary line (one physical line not exceeding 80 characters) terminated by a period, question mark, or exclamation point. When writing more (encouraged), this must be followed by a blank line, followed by the rest of the docstring starting at the same cursor position as the first quote of the first line.

Also see: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html


`ni-python-styleguide lint` probably warns about this.

Reply:
Updated the description based on the google style guide